base/error.jl

```diff
@@ -52,28 +52,31 @@ end
 
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
+    retry(f, [on_error]; n=1; max_delay=0) -> Function
```

Comment:
`condition` seemed too close to condition variables constructed by `Condition()`. `on_error` could be read as "retry on error".